x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -8,8 +8,10 @@ import (
 	"google.golang.org/grpc/codes"
```

Comment:
I'm sure you and @bryanchriswhite have discussed & thought through this, but don't we have issues are slashing misbehaving suppliers if they do this mid session?

Lots of nuances, edges cases and details that I'm omitting above.

Reply:
Good call!

`MinStake` should always be enough to pay for any slashing. Otherwise a malicious `Supplier` could attack the network by staking `MinStake`.